test/unit/taxonomy/publishing_api_root_taxon_parser_test.rb

```diff
@@ -0,0 +1,105 @@
+require 'test_helper'
+
+class Taxonomy::PublishingApiRootTaxonParserTest < ActiveSupport::TestCase
+  def parsed_result(expanded_links)
+    Taxonomy::PublishingApiRootTaxonParser.parse_taxons(expanded_links)
+  end
+
+  # Example of expanded links hash:
+  #
+  # {
+  #   "expanded_links" => {
+  #     "child_taxons" => [
+  #       {
+  #         "base_path" => "/root-path",
+  #         "content_id" => "c58fdadd-7743-46d6-9629-90bb3ccc4ef0",
+  #         "title" => "I am the root taxon.",
+  #         "links" => {
+  #           "child_taxons" => [
+  #             {
+  #               "base_path" => "/child-path",
+  #               "content_id" =>"47b6ce42-0bfa-42ee-9ff1-7a9c71ee9727",
+  #               "title" => "I am the child taxon.",
+  #               "links" => {}
+  #             }
+  #           ]
+  #         }
+  #       }
+  #     ]
+  #   }
+  # }
+
+  test ".parse_taxons returns an empty array when there are no child taxons" do
+    assert parsed_result({ "expanded_links" => {} }).empty?
+  end
+
+  test ".parse_taxons parses a single child_taxon" do
```

Comment:
The tests here might read easier if the test values are inlined, so you have all the information you need inside the `test` block:

```ruby
test ".parse_taxons parses a single child_taxon" do
   single_root_node = {
    "expanded_links" => {
      "child_taxons" => [
        node
      ]
    }
  }

  result = parsed_result(single_root_node)
   
  assert is_an_array_of_taxons(result)
  assert result.length == 1
end
```